### PR TITLE
Basket-wishlist

### PR DESCRIPTION
**Pull request by animeshgarai09**
This pr consists of the below features.

1. Restricted paths i.e wishlist and cart not accessible before login
2. wishlist management and cart management.

Guest login is enabled you can quickly login through it.

### CODE REVIEW

**Review by partha8 — APPROVED** [review]
Everything looks great, can't suggest any changes.

File: src/pages/wishlist/wishlist.js [hunk 1]
```diff
@@ -1,38 +1,52 @@
...
 
 const Wishlist = () => {
+
+    const { WishlistState, removeFromWishlist } = useWishlist()
+    const { addToBasket } = useBasket()
+
+    const handleMoveToBasket = (item) => () => {
+        removeFromWishlist(item._id)
+        addToBasket(item)
+    }
```
Comment (partha8) [inline src/pages/wishlist/wishlist.js:15]: Something new for me, the syntax of the function.

Comment (animeshgarai09) [reply src/pages/wishlist/wishlist.js:15]: This is a currying function.

File: src/index.js [hunk 2]
```diff
@@ -4,20 +4,26 @@ import App from './App';
...
-        <AuthProvider>
+        <ToastProvider>
             <ProductProvider>
-                <ToastProvider>
-                    <App />
-                </ToastProvider>
+                <BasketProvider>
+                    <WishlistProvider>
+                        <AuthProvider>
+                            <App />
```
Comment (anik31) [inline src/index.js:21]: appreciation - nice, you have used different context for cart, wishlist & auth

File: src/store/providers/basket-provider.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            ).then((response) => {
+
+                setBasketState(response.data.cart);
+                toast({
+                    status: "success",
+                    message: "Item added in your basket",
+                    duration: 2
+                })
+            }).catch((err) => {
+                console.log(err)
```
Comment (anik31) [inline src/store/providers/basket-provider.js:32]: suggestion - you can use try catch here

File: src/store/providers/basket-provider.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    //             `/api/user/cart/${id}`,
+    //             {
+    //                 qty: quantity,
+    //             },
+    //             {
+    //                 headers: {
+    //                     authorization: encodedToken,
+    //                 },
+    //             }
+    //         );
```
Comment (anik31) [inline src/store/providers/basket-provider.js:68]: suggestion - avoid pushing commented code

File: src/hooks/index.js [hunk 5]
```diff
@@ -10,4 +12,27 @@ export const useInputHandler = (state) => {
...
     return { inputState, inputChange }
+}
+
+export const useSetUserDetails = () => {
+    const { AuthDispatcher } = useAuth()
+    const { setBasketState } = useBasket()
+    const { setWishlistState } = useWishlist()
+
+    return (user, token) => {
+        console.log(user)
```
Comment (nayyyhaa) [inline src/hooks/index.js:23]: can remove console.log()'s from the code if not required anymore.

**Review by anik31 — APPROVED** [review]
Code quality is very good. And the functionality is working nicely in the preview. Nice work

**Review by nayyyhaa — APPROVED** [review]
Appreciation: Well-written code. Folder structure, variable naming + all the concepts are used perfectly! only few console.logs & commented line to be removed, rest looks good!

**Pull request merged**